torchaudio/csrc/sox_utils.cpp

```diff
@@ -99,9 +99,6 @@ void validate_input_file(const SoxFormat& sf) {
   if (sf->encoding.encoding == SOX_ENCODING_UNKNOWN) {
     throw std::runtime_error("Error loading audio file: unknown encoding.");
   }
-  if (sf->signal.length == 0) {
-    throw std::runtime_error("Error reading audio file: unkown length.");
-  }
```

Comment:
This error is removed in preparation of streaming, or because the length of the signal when visiting the bytes path is zero?

Reply:
It's the later. For container formats, which does not have a length information in header, these values become zero.

Reply:
thx for clarifying!